File: ReverseLinkedList/ReverseLinkedList.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
        Stack<ListNode> stk = new Stack<ListNode>();
        while (head != null) {
            stk.Push(head);
            head = head.next;
        }
        
        head = null;
        ListNode cur = null;
        while (stk.Any()) {
            ListNode tmp = stk.Pop();
            
            if (cur == null) {
                head = tmp;
            } else {
                cur.next = tmp;
            }
                
            cur = tmp;
        }
        if (cur != null) {
            cur.next = null;
        }
        
        return head;
    }
}

File: ReverseLinkedList/ReverseLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        stk = []
        while head is not None:
            stk.append(head)
            head = head.next
            
        head = None
        cur = None
        while len(stk) > 0:
            tmp = stk.pop()
            
            if cur is None:
                cur = tmp
                head = tmp
            else:
                cur.next = tmp
                cur = cur.next
                
        if cur is not None:
            cur.next = None
        
        return head